test/ha/resilience_test.go

```diff
@@ -0,0 +1,169 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ha
+
+import (
+	"context"
+	"strconv"
+	"sync"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/watch"
+	"knative.dev/serving/pkg/apis/autoscaling"
+	"knative.dev/serving/pkg/apis/serving"
+	rtesting "knative.dev/serving/pkg/testing/v1"
+	"knative.dev/serving/test"
+	"knative.dev/serving/test/e2e"
+	v1test "knative.dev/serving/test/v1"
+)
+
+const (
+	minimumNumberOfReplicas = 2
+	maximumNumberOfReplicas = 2
+)
+
+func TestActivatorNotInRequestPath(t *testing.T) {
+	clients := e2e.Setup(t)
+	ctx := context.Background()
+
+	// Create first service that we will continually probe during disruption scenario.
+	names, resources := createPizzaPlanetService(t,
+		rtesting.WithConfigAnnotations(map[string]string{
+			autoscaling.MinScaleAnnotationKey:  strconv.Itoa(minimumNumberOfReplicas), // Make sure we don't scale to zero during the test.
+			autoscaling.MaxScaleAnnotationKey:  strconv.Itoa(maximumNumberOfReplicas),
+			autoscaling.TargetBurstCapacityKey: "0", // The Activator is only added to the request path during scale from zero scenarios.
+		}),
+	)
+	test.EnsureTearDown(t, clients, &names)
+
+	testUptimeDuringUserPodDeletion(t, ctx, clients, names, resources)
+}
+
+func TestActivatorInRequestPathAlways(t *testing.T) {
+	clients := e2e.Setup(t)
+	ctx := context.Background()
+
+	// Create first service that we will continually probe during disruption scenario.
+	names, resources := createPizzaPlanetService(t,
+		rtesting.WithConfigAnnotations(map[string]string{
+			autoscaling.MinScaleAnnotationKey:  strconv.Itoa(minimumNumberOfReplicas), // Make sure we don't scale to zero during the test.
+			autoscaling.MaxScaleAnnotationKey:  strconv.Itoa(maximumNumberOfReplicas),
+			autoscaling.TargetBurstCapacityKey: "-1", // Make sure all requests go through the activator.
+		}),
+	)
+	test.EnsureTearDown(t, clients, &names)
+
+	testUptimeDuringUserPodDeletion(t, ctx, clients, names, resources)
+}
+
+func TestActivatorInRequestPathPossibly(t *testing.T) {
+	clients := e2e.Setup(t)
+	ctx := context.Background()
+
+	// Create first service that we will continually probe during disruption scenario.
+	names, resources := createPizzaPlanetService(t,
+		rtesting.WithConfigAnnotations(map[string]string{
+			autoscaling.MinScaleAnnotationKey:  strconv.Itoa(minimumNumberOfReplicas), // Make sure we don't scale to zero during the test.
+			autoscaling.MaxScaleAnnotationKey:  strconv.Itoa(maximumNumberOfReplicas),
+			autoscaling.TargetBurstCapacityKey: "1", // The Activator may be in the path, depending on the revision scale and load.
```

Comment:
Can you comment why you picked 1?

Reply:
I hadn't thought about that; I simply set it to a number other than 0 or -1. However, upon checking, I noticed that the default value is 200. Should I go with the default instead?

Reply:
using default instead